AVInput/AVInput.cpp

```diff
@@ -224,5 +327,923 @@ string AVInput::currentVideoMode(bool &success)
+    }
+}
+
+std::string AVInput::getRawSPD(int iPort)
+{
+    LOGINFO("AVInput::getSPDInfo");
+    vector<uint8_t> spdVect({'u','n','k','n','o','w','n' });
+    std::string spdbase64 = "";
+    try {
+        LOGWARN("AVInput::getSPDInfo");
+        vector<uint8_t> spdVect2;
+        device::HdmiInput::getInstance().getHDMISPDInfo(iPort, spdVect2);
+        spdVect = spdVect2;//edidVec must be "unknown" unless we successfully get to this line
+
+        //convert to base64
+        uint16_t size = min(spdVect.size(), (size_t)numeric_limits<uint16_t>::max());
+
+        LOGWARN("AVInput::getSPD size:%d spdVec.size:%lu", size, spdVect.size());
+
+        if(spdVect.size() > (size_t)numeric_limits<uint16_t>::max()) {
+            LOGERR("Size too large to use ToString base64 wpe api");
+            return spdbase64;
+        }
+
+        LOGINFO("------------getSPD: ");
+        for (size_t itr =0; itr < spdVect.size(); itr++) {
+            LOGINFO("%02X ", spdVect[itr]);
+        }
+        Core::ToString((uint8_t*)&spdVect[0], size, false, spdbase64);
+    }
+    catch (const device::Exception& err) {
+        LOG_DEVICE_EXCEPTION1(std::to_string(iPort));
+    }
+    return spdbase64;
+}
+
+std::string AVInput::getSPD(int iPort)
+{
+    LOGINFO("AVInput::getSPDInfo");
+    vector<uint8_t> spdVect({'u','n','k','n','o','w','n' });
+    std::string spdbase64 = "";
+    try {
+        LOGWARN("AVInput::getSPDInfo");
+        vector<uint8_t> spdVect2;
+        device::HdmiInput::getInstance().getHDMISPDInfo(iPort, spdVect2);
+        spdVect = spdVect2;//edidVec must be "unknown" unless we successfully get to this line
+
+        //convert to base64
+        uint16_t size = min(spdVect.size(), (size_t)numeric_limits<uint16_t>::max());
+
+        LOGWARN("AVInput::getSPD size:%d spdVec.size:%lu", size, spdVect.size());
+
+        if(spdVect.size() > (size_t)numeric_limits<uint16_t>::max()) {
+            LOGERR("Size too large to use ToString base64 wpe api");
+            return spdbase64;
+        }
+
+        LOGINFO("------------getSPD: ");
+        for (size_t itr =0; itr < spdVect.size(); itr++) {
+          LOGINFO("%02X ", spdVect[itr]);
+        }
+        if (spdVect.size() > 0) {
+            struct dsSpd_infoframe_st pre;
+            memcpy(&pre,spdVect.data(),sizeof(struct dsSpd_infoframe_st));
+
+            char str[200] = {0};
+            sprintf(str, "Packet Type:%02X,Version:%u,Length:%u,vendor name:%s,product des:%s,source info:%02X",
+            pre.pkttype,pre.version,pre.length,pre.vendor_name,pre.product_des,pre.source_info);
+            spdbase64 = str;
+        }
+    }
+    catch (const device::Exception& err) {
+        LOG_DEVICE_EXCEPTION1(std::to_string(iPort));
+    }
+    return spdbase64;
+}
+
+uint32_t AVInput::setEdidVersionWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    int portId = 0;
+
+    LOGINFOMETHOD();
+    string sPortId = parameters["portId"].String();
```

Comment:
please use `parameters["portId"].Number()` if it is number

Reply:
Modified code to use Number()